### DISCUSSION TITLE
Reflecting view in MySQL database (hosted on DigitalOcean) returns empty columns

### DISCUSSION THREAD

**Discussion by lightandbright** [sqlalchemy/sqlalchemy#8902] [Usage Questions] [answered]
Hi there, Im experiencing strange behaviour when reflecting a view on MySQL database hosted on DigialOcean. Tables are reflected fine. When I try to reflect a view on that very same database I am able to get some data back (the view definition is returned fine for example), but the columns return as empty lists. Any ideas on what might be wrong? Thanks much.

`
from sqlalchemy import create_engine, select
`
`
from sqlalchemy import inspect
`
`
from sqlalchemy import MetaData
`

`
e = create_engine('mysql+mysqldb://admin:some_path.ondigitalocean.com:25060/defaultdb')
`

`
with e.begin() as conn:
`
`
    conn.exec_driver_sql(
`
`
        'CREATE VIEW defaultdb.v_today (today) AS SELECT CURRENT_DATE;'
`
`
    )
`

`
db_structure = inspect(e)
`

`
print(db_structure.get_view_definition("v_today", schema="defaultdb"))
`
`
print(db_structure.get_columns("v_today", schema="defaultdb"))
`

output:
`CREATE VIEW "v_today" ("today") AS select curdate() AS "CURRENT_DATE"`
`[]`

**Comment by zzzeek** [comment, marked as answer]
hi - 

what SQLAlchemy version are you on and what is the **exact** database server string you are using?  

```
MariaDB [(none)]> SELECT VERSION();
+-----------------+
| VERSION()       |
+-----------------+
| 10.5.16-MariaDB |
+-----------------+

```

the CREATE VIEW definition does not look like standard MySQL / Mariadb, which produces SQL of this form:

```
CREATE ALGORITHM=UNDEFINED DEFINER=`scott`@`%` SQL SECURITY DEFINER VIEW `test_schema`.`v_today` (`today`) AS select curdate() AS `CURRENT_DATE`
```

the quoting style also looks like [ANSI_QUOTES](https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_ansi_quotes), which is fine, we reflect that also.

however the lack of "SQL SECURITY DEFINER" etc. looks like #8588 which was released in 1.4.42 to support MySQL variants such as StarRocks.   Are you on at least SQLAlchemy 1.4.42?

**Reply by lightandbright** [reply]
great, upgrading helped. Fyi, I was running a managed database on DO MySQL 8.0.28 and SQLAlchemy 1.4.36. After upgrade all works well, thanks very much
